src/registriesView.ts

```diff
@@ -238,17 +289,18 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
          */
 
         if (registryContext) {
-          const notChangedRegisty = registries.find((registry) => registry.Name === regName && registry.URL === regURL && registry.Secure === (secure === 'Yes'));
-          if (notChangedRegisty) {
-              return null;
-          }
-          await vscode.commands.executeCommand('openshift.componentTypesView.registry.remove', registryContext, true);
+            const notChangedRegisty = registries.find((registry) => registry.Name === regName && registry.URL === regURL && registry.Secure === (secure === 'Yes'));
+            if (notChangedRegisty) {
+                return null;
+            }
+            await vscode.commands.executeCommand('openshift.componentTypesView.registry.remove', registryContext, true);
         }
 
         const newRegistry = await OdoImpl.Instance.addRegistry(regName, regURL, token);
         ComponentTypesView.instance.addRegistry(newRegistry);
 
-        RegistryViewLoader.refresh();
+        ComponentTypesView.instance.getAllComponents();
+        RegistryViewLoader.refresh()
```

Comment:
This introduces cycle dependency. RegistryViewLoader depends on ComponentTypesView and vice versa.

Reply:
RegistryViewLoader is going to be refreshed during `ComponentTypesView.instance.getAllComponents()` call, because it sends 'refresh' event through the subject.